osu.Game/Screens/Select/Details/AdvancedStats.cs

```diff
@@ -149,6 +151,13 @@ private void updateStatistics()
                 }
             }
 
+            // update tooltip anyway
+            else if (baseDifficulty != null)
```

Comment:
does this have to be handled out-of-band like this? can't it be fixed by removing the `mods.Value.Any(...)` checks from the `if` above?

seems simpler that way.

Reply:
to elaborate, my fix on top of `master` would be just to

```diff
diff --git a/osu.Game/Screens/Select/Details/AdvancedStats.cs b/osu.Game/Screens/Select/Details/AdvancedStats.cs
index 2a9e69f141..f56e84abce 100644
--- a/osu.Game/Screens/Select/Details/AdvancedStats.cs
+++ b/osu.Game/Screens/Select/Details/AdvancedStats.cs
@@ -127,8 +127,7 @@ private void updateStatistics()
 
             IRulesetInfo ruleset = gameRuleset?.Value ?? beatmapInfo.Ruleset;
 
-            if (baseDifficulty != null &&
-                (mods.Value.Any(m => m is IApplicableToDifficulty) || mods.Value.Any(m => m is IApplicableToRate)))
+            if (baseDifficulty != null)
             {
                 BeatmapDifficulty originalDifficulty = new BeatmapDifficulty(baseDifficulty);
 

```

to my eye the code inside the block handles no difficulty / rate-changing mods just fine?

Reply:
I've done the same thing and I somehow get an unreproducible bug where AR hasn't updated when I've switched between difficulties with DT enabled
When I relaunched the game without any changes - it worked correctly
I don't know any possible reason for this

Reply:
to me that sounds like a completely separate bug and there is no explanation for why switching from that initial variant to what is in this PR right now would fix it...?

Reply:
Yes, it's a separate bug
I found a way to reproduce it - just change anything in the code and after the first launch it will happen

Reply:
> just change anything in the code

i'm not sure how this a reproduction scenario but sure.

at this point if you are able, please fix it in this pull, just to save on the ceremony.